src/vpnconf/vpnconf.py

```diff
@@ -105,40 +223,61 @@ def main():
                 error="Possible values for --log-level are "
                 + "debug, info, warning, error, and critical.",
             ),
+            Optional(Or("-e", "--extra-routes")): list,
+            "--no-ipv4": bool,
+            "--no-ipv6": bool,
             str: object,  # Don't care about other keys, if any
         }
     )
 
     try:
+        # Validate the input arguments against the schema
         args = schema.validate(args)
     except SchemaError as err:
-        # Exit because one or more of the arguments were invalid
+        # If arguments validation fails, print the error and terminate the program
         print(err, file=sys.stderr)
         sys.exit(1)
 
-    # Assign validated arguments to variables
+    # Assign the validated arguments to corresponding variables
     filename = args["<filename>"]
     log_level = args["--log-level"]
+    no_ipv4 = args["--no-ipv4"]
+    no_ipv6 = args["--no-ipv6"]
 
-    # Set up logging
+    # Configure the logging system
     logging.basicConfig(
         format="%(asctime)-15s %(levelname)s %(message)s", level=log_level.upper()
     )
 
+    # Read the configuration file and remove any existing route lines
     lines = read_config_no_routes(filename)
 
+    # Generate a UUID and use it to get the IP endpoints for the specified instance
     guid = uuid.uuid1()
     ip_set = get_endpoints(guid, INSTANCE)
 
-    # add our static routes
-    ip_set = ip_set.union(STATIC_ROUTES)
-    lines += generate_routes(ip_set)
+    # If extra routes files are provided, parse them and add their routes to the set
+    extra_routes, dns_errors = parse_extra_routes(args["--extra-routes"])
+    if dns_errors:
+        # If there were unresolved hostnames, log an error and terminate the program
+        logging.error(
+            f"Could not resolve the following hostnames: {', '.join(dns_errors)}"
+        )
```

Comment:
Do you need to shutdown logging even when you exit uncleanly?
```suggestion
        )
        logging.shutdown()
```